os_migrate/roles/import_workloads/tasks/workload.yml

```diff
@@ -34,6 +34,51 @@
       dst_filters: "{{ os_migrate_dst_filters }}"
     when: prelim.changed
 
+  # Stop VM if not stopped and if config set to true
+  - name: perform workload stop if os_migrate_workload_stop_before_migration is true
+    openstack.cloud.server_action:
+      auth: "{{ os_migrate_src_auth }}"
+      auth_type: "{{ os_migrate_src_auth_type|default(omit) }}"
+      region_name: "{{ os_migrate_src_region_name|default(omit) }}"
+      server: "{{ item }}"
+      action: stop
+      wait: yes
+      timeout: "{{ os_migrate_timeout }}"
+      validate_certs: "{{ os_migrate_src_validate_certs|default(omit) }}"
+      ca_cert: "{{ os_migrate_src_ca_cert|default(omit) }}"
+      client_cert: "{{ os_migrate_src_client_cert|default(omit) }}"
+      client_key: "{{ os_migrate_src_client_key|default(omit) }}"
+    when:
+      - prelim.changed
+      - os_migrate_workload_stop_before_migration
+
+  # Confirm that VM is now is expected state
+  - name: gather info on stopped workload
```

Comment:
Whether the VM is indeed stopped will be checked by the `import_workload_src_check` module, so we don't need to double that i think.

Reply:
Thanks. Changes made in https://github.com/os-migrate/os-migrate/pull/464/commits/f551ffb9d0f5972b88eea01ddabeebc97421cb77